pkg/apis/serving/v1/revision_helpers.go

```diff
@@ -87,6 +107,41 @@ func (rs *RevisionSpec) GetContainer() *corev1.Container {
 	return &corev1.Container{}
 }
 
+// SetRoutingState sets the routingState label on this Revision and updates the
+// routingStateModified annotation.
+func (r *Revision) SetRoutingState(state RoutingState) {
+	stateStr := string(state)
+	t := r.ObjectMeta.Annotations[serving.RoutingStateModifiedAnnotationKey]
+	if t != "" && r.Labels[serving.RoutingStateLabelKey] == stateStr {
+		return // Don't update timestamp if no change
+	}
+
+	if r.Labels == nil {
+		r.Labels = make(map[string]string, 1)
+	}
+	r.Labels[serving.RoutingStateLabelKey] = stateStr
+
+	if r.Annotations == nil {
+		r.Annotations = make(map[string]string, 1)
+	}
+	r.Annotations[serving.RoutingStateModifiedAnnotationKey] = fmt.Sprint(time.Now().Format(time.RFC3339))
```

Comment:
1 time.Format is already a string
2. Why not just `time.NowUnix()` converted to string?

Reply:
NowUnix() just gives seconds from epoch - it's not very human readable. I think it will be easier to debug the resource if its in an easy-to-read string format.

https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.18

It appears that RFC3339 is the standard timestamp format that canonical k8s is using, so I think it makes sense to keep consistency.

Reply:
More work for the computers, but sure :)

Reply:
Remove the `fmt.Sprint` nonetheless.